packages/platform/types/event-types/event-types_2024_06_14/inputs/create-event-type.input.ts

```diff
@@ -22,27 +23,32 @@ import {
 } from "./booking-limits-duration.input";
 import type { BookingWindow_2024_06_14 } from "./booking-window.input";
 import { ValidateBookingWindow } from "./booking-window.input";
+import { EventTypeColor_2024_06_14 } from "./event-type-color.input";
 import { ValidateLocations_2024_06_14 } from "./locations.input";
 import type { Location_2024_06_14 } from "./locations.input";
 import { Recurrence_2024_06_14 } from "./recurrence.input";
+import { RequiresConfirmation_2024_06_14, ValidateRequiresConfirmation } from "./requires-confirmation.input";
+import { Seats_2024_06_14 } from "./seats.input";
 
 export const CREATE_EVENT_LENGTH_EXAMPLE = 60;
 export const CREATE_EVENT_TITLE_EXAMPLE = "Learn the secrets of masterchief!";
 export const CREATE_EVENT_DESCRIPTION_EXAMPLE =
   "Discover the culinary wonders of the Argentina by making the best flan ever!";
 
-export class CreateEventTypeInput_2024_06_14 {
+export class BaseCreateEventTypeInput_2024_06_14 {
+  @IsOptional()
   @IsInt()
   @Min(1)
-  @DocsProperty({ example: CREATE_EVENT_LENGTH_EXAMPLE })
-  lengthInMinutes!: number;
+  lengthInMinutes?: number;
 
+  @IsOptional()
   @IsString()
   @DocsProperty({ example: CREATE_EVENT_TITLE_EXAMPLE })
-  title!: string;
+  title?: string;
 
+  @IsOptional()
   @IsString()
-  slug!: string;
+  slug?: string;
```

Comment:
These fields were made optional in the base class because it serves as a foundation for various derived classes with different requirements. In `CreateTeamEventTypeInput_2024_06_14`, the specific input class where these fields are essential, they are overridden to be required.

This approach allows us to enforce stricter validation rules in derived classes, as loosening restrictions in derived classes is not permissible. Since we’re also extending `UpdateEventTypeInput_2024_06_14` from this base class, where these fields need to remain optional, the base class must accommodate this by making them optional. This strategy provides flexibility across different use cases while ensuring that necessary fields are mandatory where needed.